src/consumer/kafka.rs

```diff
@@ -71,6 +82,56 @@ pub async fn start_consumer(
     .await
 }
 
+async fn create_missing_topics(
+    kafka_client_config: ClientConfig,
+    topics: &[&str],
+    default_topic_partitions: i32,
+) -> Result<(), Error> {
+    let admin_client: AdminClient<_> = kafka_client_config.create().unwrap();
+    let mut topics_to_create = Vec::new();
+    for topic in topics {
+        // Weirdly, this call will actually create the topic if it doesn't exist. So it needs to be overridden in the next steps.
+        let topic_metadata: Metadata = admin_client
+            .inner()
+            .fetch_metadata(Some(topic), Duration::new(5, 0))
+            .expect("Failed to fetch metadata");
+
+        if !topic_metadata.topics().is_empty() {
+            for meta_topic in topic_metadata.topics() {
+                if meta_topic.error().is_some()
+                    && meta_topic.partitions().len() as i32 != default_topic_partitions
```

Comment:
Do we need to create the topic if `meta_topic` has more partitions than `default_topic_partitions`?

Reply:
If the topic already exists and has partitions, then the `error` will be null and this check won't get hit.